modules/soc_i3/main.sh

```diff
@@ -66,46 +67,28 @@ incrementTimer(){
 soctimer=$(<$soctimerfile)
 openwbDebugLog ${DMOD} 1 "Lp$CHARGEPOINT: timer = $soctimer"
 cd /var/www/html/openWB/modules/soc_i3
-if (( soctimer < 60 )); then
+if (( soctimer < (6 * intervall) )); then
 	openwbDebugLog ${DMOD} 1 "Lp$CHARGEPOINT: Nothing to do yet. Incrementing timer."
 	incrementTimer
 else
 	openwbDebugLog ${DMOD} 1 "Lp$CHARGEPOINT: Requesting SoC"
 	echo 0 > $soctimerfile
-	re='^-?[0-9]+$'
-	abfrage=$(sudo php index.php --chargepoint=$CHARGEPOINT --username=$user --password=$pass --vin=$vin | jq '.')
-	soclevel=$(echo $abfrage | jq '.chargingLevel')
-	if  [[ $soclevel =~ $re ]] ; then
-		if (( $soclevel != 0 )) ; then
-			echo $soclevel > $socfile
-		fi
-	fi
-	openwbDebugLog ${DMOD} 1 "Lp$CHARGEPOINT: SoC: $soclevel"
+	
+	ARGSFILE="$RAMDISKDIR/soc_${SOCMODULE}_lp${CHARGEPOINT}_args"
```

Comment:
Die Idee mit der Datei in der Ramdisk stellt potentiell ein Sicherheitsrisiko dar, weil die Zugangsdaten dann für jeden zugänglich sind. Was spricht dagegen, dem Python Skript die Parameter direkt mitzugeben? Gerne auch mir "argparse", damit es einfacher zu lesen und unabhängig von der Reihenfolge ist.

Reply:
Ich hatte das aus dem Kia-Modul kopiert, wo es einfach zu unübersichtilch wurde. 
Das JSON wird jetzt BASE64-kodiert als Parameter übergeben (zwecks Umlauten u.ä.) und es findet keine Zwischenspeicherung in der Ramdisk mehr statt. Für Kia habe ich einen getrennten PR mit derselben Änderung aufgemacht.